spring-cloud-generator/scripts/sanity-checks.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# This script performs the following sanity checks for the library generation
```

Comment:
@meltsufin I'm leaving a couple notes to indicate the reach of these checks.

We have the empty library list [check](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/07a3a1c6f8cbf66c9f488d447c75cec435a941ab/spring-cloud-generator/scripts/sanity-checks.sh#L22) to prevent the case where an empty list still gives a successful run of the generator. 

Then we have the [check](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/07a3a1c6f8cbf66c9f488d447c75cec435a941ab/spring-cloud-generator/scripts/sanity-checks.sh#L53) to see if the generated files are present in our previews folder (checks pom, classes, etc are present)

Since we [start with an empty previews folder](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/07a3a1c6f8cbf66c9f488d447c75cec435a941ab/.github/workflows/generateAutoConfigs.yaml#L69-L74), confirming the library list is not empty and checking that the generated source files are present **we can confirm the generator was run**. In other words, these two checks help confirm at least one library was successfully generated and written in the previews folder.

Reply:
Thanks! Perfect!